x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/ClientTransformIndexer.java

```diff
@@ -440,6 +450,134 @@ SeqNoPrimaryTermAndIndex getSeqNoPrimaryTermAndIndex() {
         return seqNoPrimaryTermAndIndex.get();
     }
 
+    @Override
+    protected void afterFinishOrFailure() {
+        closePointInTime();
+        super.afterFinishOrFailure();
+    }
+
+    @Override
+    protected void onStop() {
+        closePointInTime();
+        super.onStop();
+    }
+
+    private void closePointInTime() {
+        if (pit == null) {
+            return;
+        }
+
+        String oldPit = pit.getEncodedId();
+        pit = null;
+        ClosePointInTimeRequest closePitRequest = new ClosePointInTimeRequest(oldPit);
+        ClientHelper.executeWithHeadersAsync(
+            transformConfig.getHeaders(),
+            ClientHelper.TRANSFORM_ORIGIN,
+            client,
+            ClosePointInTimeAction.INSTANCE,
+            closePitRequest,
+            ActionListener.wrap(response -> { logger.trace("[{}] closed pit search context [{}]", getJobId(), oldPit); }, e -> {
+                // note: closing the pit should never throw, even if the pit is invalid
+                logger.error(new ParameterizedMessage("[{}] Failed to close point in time reader", getJobId()), e);
+            })
```

Comment:
I think the `logger.trace` should have a message supplier like `() -> new ParameterizedMessage` to prevent strings from being created when `trace` is disabled.

Not a huge deal as this is not a "hot path"

Reply:
I don't think this applies. This is only a problem if one or more arguments needs to be constructed, e.g. if `getJobId()` would build the id and therefore execute something. This is not the case.

The message string itself gets only constructed _after_ the check whether trace is enabled or not.

I wish we have static code analysis for this this, it's such a common problem.